src/proof/bitvector_proof.cpp

```diff
@@ -9,105 +9,44 @@
-#include "theory/bv/theory_bv_rewrite_rules.h"
-
-using namespace CVC4::theory;
-using namespace CVC4::theory::bv;
 
 namespace CVC4 {
-
 BitVectorProof::BitVectorProof(theory::bv::TheoryBV* bv,
                                TheoryProofEngine* proofEngine)
     : TheoryProof(bv, proofEngine),
       d_declarations(),
       d_seenBBTerms(),
       d_bbTerms(),
       d_bbAtoms(),
-      d_resolutionProof(NULL),
-      d_cnfProof(NULL),
-      d_isAssumptionConflict(false),
-      d_bitblaster(NULL),
-      d_useConstantLetification(false) {}
-
-void BitVectorProof::initSatProof(CVC4::BVMinisat::Solver* solver) {
-  Assert (d_resolutionProof == NULL);
-  d_resolutionProof = new BVSatProof(solver, &d_fakeContext, "bb", true);
-}
-
-theory::TheoryId BitVectorProof::getTheoryId() { return theory::THEORY_BV; }
-void BitVectorProof::initCnfProof(prop::CnfStream* cnfStream,
-                                  context::Context* cnf) {
-  Assert (d_resolutionProof != NULL);
-  Assert (d_cnfProof == NULL);
-  d_cnfProof = new LFSCCnfProof(cnfStream, cnf, "bb");
-
-  // true and false have to be setup in a special way
-  Node true_node = NodeManager::currentNM()->mkConst<bool>(true);
-  Node false_node = NodeManager::currentNM()->mkConst<bool>(false).notNode();
-
-  d_cnfProof->pushCurrentAssertion(true_node);
-  d_cnfProof->pushCurrentDefinition(true_node);
-  d_cnfProof->registerConvertedClause(d_resolutionProof->getTrueUnit());
-  d_cnfProof->popCurrentAssertion();
-  d_cnfProof->popCurrentDefinition();
-
-  d_cnfProof->pushCurrentAssertion(false_node);
-  d_cnfProof->pushCurrentDefinition(false_node);
-  d_cnfProof->registerConvertedClause(d_resolutionProof->getFalseUnit());
-  d_cnfProof->popCurrentAssertion();
-  d_cnfProof->popCurrentDefinition();
-}
-
-void BitVectorProof::setBitblaster(bv::TBitblaster<Node>* bb) {
-  Assert (d_bitblaster == NULL);
-  d_bitblaster = bb;
-}
-
-BVSatProof* BitVectorProof::getSatProof() {
-  Assert (d_resolutionProof != NULL);
-  return d_resolutionProof;
+      d_bitblaster(nullptr),
+      d_useConstantLetification(false),
+      d_cnfProof()
+{
 }
 
-void BitVectorProof::registerTermBB(Expr term) {
-  Debug("pf::bv") << "BitVectorProof::registerTermBB( " << term << " )" << std::endl;
-
-  if (d_seenBBTerms.find(term) != d_seenBBTerms.end())
-    return;
-
-  d_seenBBTerms.insert(term);
-  d_bbTerms.push_back(term);
-
-  // If this term gets used in the final proof, we will want to register it. However,
-  // we don't know this at this point; and when the theory proof engine sees it, if it belongs
-  // to another theory, it won't register it with this proof. So, we need to tell the
-  // engine to inform us.
-
-  if (theory::Theory::theoryOf(term) != theory::THEORY_BV) {
-    Debug("pf::bv") << "\tMarking term " << term << " for future BV registration" << std::endl;
-    d_proofEngine->markTermForFutureRegistration(term, theory::THEORY_BV);
-  }
+std::string BitVectorProof::getBBTermName(Expr expr)
```

Comment:
It seems like this method and at least `registerTermBB` were moved, which makes the diff unnecessarily large/complicated. It would be great if this change could be reverted.